connection.js

```diff
@@ -985,6 +986,7 @@ Connection.prototype.rcpt_ok_respond = function (retval, msg) {
         this.results.add(this, {err: 'rcpt_ok_respond found no transaction'});
         return;
     }
+    if (!msg) msg = this.last_rcpt_msg;
     var rcpt = this.transaction.rcpt_to[this.transaction.rcpt_to.length - 1];
```

Comment:
No - this is a V8 optimization.  Over the life of the Haraka process we create thousands of connection objects and creating each object with identical properties prevents V8 from creating a hidden class.  See https://developers.google.com/v8/design

There's also a paper from a talk about this about v8 optimizations but I can't find it right now.

I also think this is quite developer friendly because one look at the constructor tells you all the properties it might have.